generators/app/templates/src/StarterKit/ServiceAgents/_base/AgentBase.cs

```diff
@@ -1,127 +1,238 @@
         {
-            using (HttpContent contentPost = new ByteArrayContent(Utf8Json.JsonSerializer.Serialize(item)))
-            {
-                using (var response = await HttpClient.PostAsync(requestUri, contentPost))
-                    return await ParseResult<T>(response);
-            }
-        }
+          Title = "Json parse error exception.",
+          Status = (int)response.StatusCode,
+          ExtraInfo = new Dictionary<string, IEnumerable<string>> { { "json", new[] { errorJson } } }
+        };
+      }
 
-        protected async Task<TReponse> PostAsync<TRequest, TReponse>(string requestUri, TRequest item)
-        {
-            using (HttpContent contentPost = new ByteArrayContent(Utf8Json.JsonSerializer.Serialize(item)))
-            using (var response = await HttpClient.PostAsync(requestUri, contentPost))
-                return await ParseResult<TReponse>(response);
-        }
+      // Throw proper exception based on HTTP status
+      throw response.StatusCode switch
+      {
+        HttpStatusCode.NotFound => new NotFoundException(message: errorResponse?.Title ?? "Not found",
+            code: errorResponse?.Code ?? "NFOUND001", messages: errorResponse?.ExtraInfo),
+        HttpStatusCode.BadRequest => new ValidationException(message: errorResponse?.Title ?? "Bad request",
+            code: errorResponse?.Code ?? "UNVALI001", messages: errorResponse?.ExtraInfo),
+        HttpStatusCode.Unauthorized => new UnauthorizedException(
+            message: errorResponse?.Title ?? "Access denied", code: errorResponse?.Code ?? "UNAUTH001",
+            messages: errorResponse?.ExtraInfo),
+        HttpStatusCode.Forbidden => new ForbiddenException(message: errorResponse?.Title ?? "Forbidden",
+            code: errorResponse?.Code ?? "FORBID001", messages: errorResponse?.ExtraInfo),
+        HttpStatusCode.BadGateway => new BadGatewayException(message: "The server could not be located",
+            code: "GTWAY001", messages: errorResponse?.ExtraInfo),
+        HttpStatusCode.GatewayTimeout => new GatewayTimeoutException(
+            message: "The connection to the server timed out", code: "GTWAY002",
+            messages: errorResponse?.ExtraInfo),
+        _ => new ServiceAgentException(message: errorResponse?.Title,
+            code: errorResponse?.Code ?? $"Status: {response.StatusCode}", messages: errorResponse?.ExtraInfo)
+      };
+    }
 
-        protected async Task<T> PutAsync<T>(string requestUri, T item)
-        {
-            using (HttpContent contentPost = new ByteArrayContent(Utf8Json.JsonSerializer.Serialize(item)))
-            {
-                using (var response = await HttpClient.PutAsync(requestUri, contentPost))
-                {
-                    return await ParseResult<T>(response);
-                }
-            }
-        }
+    protected async Task<T> GetAsync<T>(string requestUri)
+    {
+      await _requestHeaderHelper.ValidateAuthHeaders(Client, Settings);
```

Comment:
Dit kan mogelijk ook beter in een Delegating Handler blijven zitten. Dan kan er per ServiceAgent enkel de nodige delegating handler (wel of niet OAuth, specifieke headers, ...) voor authenticatie toegevoegd worden en moet dit niet via config onderhouden worden.